vexim/siteaddsubmit.php

```diff
@@ -78,13 +72,31 @@
     $smtphomepath = $domainpath . "/" . $_POST['localpart'] . "/Maildir";
     $pophomepath = $domainpath . "/" . $_POST['localpart'];
   }
-//Gah. Transactions!! -- GCBirzan
-  if ((validate_password($_POST['clear'], $_POST['vclear'])) &&
-    ($_POST['type'] != "alias")) {
-      if (!password_strengthcheck($_POST['clear'])) {  
-        header ("Location: site.php?weakpass={$_POST['domain']}");
-        die;
-      }
+
+  if ($_POST['type'] === "alias") {
+    $query = "INSERT INTO domainalias (domain_id, alias)
+              SELECT domains.domain_id, :alias FROM domains WHERE domains.domain_id=:domain_id";
+    $sth = $dbh->prepare($query);
+    $sth->execute(array(':domain_id'=>$_POST['aliasdest'], ':alias'=>$_POST['domain']));
+    if ($sth->rowCount()!==1) {
+      header ("Location: site.php?failaddeddomerr={$_POST['domain']}");
+      die;
+    } else {
+      header ("Location: site.php?added={$_POST['domain']}" .
+              "&amp;type={$_POST['type']}");
```

Comment:
I know, why we put it. There was an error in HTML checks.
https://www.w3.org/TR/html4/appendix/notes.html#h-B.2.2